rero_ils/modules/deduplications/api.py

```diff
@@ -0,0 +1,430 @@
+
+        # build a dict with type: date
+        prov = get_editor(data)
+        new_candidates = []
+        for candidate in candidates:
+            # build a dict with type: date for each candidate
+            c_prov = get_editor(candidate._source.to_dict())
+            if (not prov and not c_prov) or (c_prov == prov):
+                new_candidates.append(candidate)
+        return new_candidates
+
+    def get_identifiers_candidates(self, data):
+        """Get the candidate list based on the identifiers.
+
+        :param data: the data to import.
+        """
+        # no candidates
+        if not data.get("identifiedBy"):
+            return []
+        # at least one identifier should match
+        criteria = Q("match_none")
+        for identifier in data["identifiedBy"]:
+            criteria |= Q("term", nested_identifiers__type=identifier["type"]) & Q(
+                "term", nested_identifiers__value__raw=identifier["value"]
+            )
+        search = self.search.filter("nested", path="nested_identifiers", query=criteria)
+
+        # should have the same main document type
+        if types := [t["main_type"] for t in data["type"]]:
+            search = search.filter("terms", type__main_type=types)
+        candidates = [hit for hit in search.execute().hits.hits]
+
+        # should match all provision activity dates
+        candidates = self.check_date(data, candidates)
+
+        # first author should match
+        candidates = self.check_responsibility_statement(data, candidates)
+
+        # editor should correspond
+        candidates = self.check_editor(data, candidates)
+
+        # scores are always one if the candidate exists (exact match)
+        return [
+            (
+                hit._source.pid,
+                hit._source.to_dict(),
+                1.0,
+                dict(
+                    main_type="1.0",
+                    identifier="1.0",
+                    reponsibility_statement="1.0",
+                    provision_activity="1.0",
+                    publication_date="1.0",
+                ),
+            )
+            for hit in candidates
+        ]
+
+    @classmethod
+    def normalize(cls, value):
+        """Normalize a given text value.
+
+        1. map each UTF-8 to the closest ascii value
+        2. transform to lower case
+        3. remove extra spaces
+        :param value: str - a given string.
+        :returns: the normalized given value
+        """
+        return re.sub(r"\s+", " ", unidecode(value).lower().strip())
+
+    def edit_distance(self, text1, text2):
+        """Compute a similarity score between two strings.
+
+        Strings are normalized and the score will be normalized between 0 an 1.
+        A score < 0.6 is replaced by 0.
+
+        :param text1: str - first string value.
+        :param text2: str - second string value.
+        :returns: [0.0-1.0] - the score: 1 means a perfect match
+        """
+        score = ratio(
```

Comment:
I would try using jaro_winkler() instead of ratio(), a few tests were convincing.